### ISSUE TITLE
Fix `selector-pseudo-class-no-unknown` false positives for nested `webkit-scrollbar` part

### ISSUE THREAD

**Issue by mrginglymus** [stylelint/stylelint#9224]
### What minimal example or steps are needed to reproduce the bug?

```css
::-webkit-scrollbar {
  &:horizontal {
    border-block-start-width: 1px;
  }
}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "selector-pseudo-class-no-unknown": true
  }
}
```

### How did you run Stylelint?

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcD4FoDuMBGBrAlgF2QGcwAnCAG3IwEMSACYAHQDs66AyeACwhJwC8IzPNXIMWbNhl6wSyDOQhgsRESQKocUPF3h0AjAAcAHgG4JdAL4tLIADQgAZjnIwActQC2cRDGNfDVwA6MEJCe3AhZwBzBBA-Q148OlhHagBXcmSmVjpGEBJMmEJ8vRzJPJBCGFcwPF5kQ2r0qAhkMHJqMORmNvTmLF7UZlK6PEKYC2tmS3NmCMhmGIAxXk9qPDjCPABPV3IcYRConGigzwArcIdYJrjyyu29msO8UfzOvGK3+wt8p-2r3aJ2iqmozCgtCg7xAn2++RsLBAliAA)

### Which Stylelint-related dependencies are you using?

```json
{
  "stylelint": "latest",
  "stylelint-config-standard": "latest"
}
```

### What did you expect to happen?

No error

### What actually happened?

```
2:4-15
error
Unknown pseudo-class selector ":horizontal"
[(selector-pseudo-class-no-unknown)](https://stylelint.io/user-guide/rules/selector-pseudo-class-no-unknown)
```

**Comment by jeddy3** [comment]
@mrginglymus I hope you had a nice holiday. Thanks for the report and for using the template.

I've labelled the issue as ready to implement.

Please consider [contributing](https://stylelint.io/contributing) if you have time. There are [steps on how to fix a bug in a rule](https://stylelint.io/contributor-guide/rules#fix-a-bug-in-a-rule) in the Contributor guide.

We could resolve the selector first, as we do with `selector-no-qualifying-type`. However, that feels a bit heavy-handed for what I assume is a seldom-seen pattern: modern nesting with a [non-standard pseudo-element](https://developer.mozilla.org/en-US/docs/Web/CSS/Reference/Selectors/::-webkit-scrollbar) when the standard scrollbar properties are [Baseline (Newly Available)](https://developer.mozilla.org/en-US/docs/Web/CSS/Reference/Properties/scrollbar-color).

It'll be an accurate fix, though.

An alternative, and possibly lighter, fix is welcome, too.

(The `selector-pseudo-class-no-unknown` rule is one of our oldest, and the conventions within it may reflect that.)
